internal/e2e/metadata_test.go

```diff
@@ -0,0 +1,189 @@
+	b, err := os.ReadFile("../api/graphql/graph/queryCollection/user/update.graphql")
+	Expect(err).To(BeNil())
+	str := string(b)
+	req := graphql.NewRequest(str)
+
+	req.Var("id", testUniqueUserId)
+	req.Var("input", map[string]string{
+		"name": testUpdatedUserName,
+	})
+
+	req.Header.Set("Cache-Control", "no-cache")
+	ctx := context.Background()
+
+	var respData struct {
+		User model.User `json:"updateUser"`
+	}
+	if err := util2.RequestWithBackoff(func() error { return client.Run(ctx, req, &respData) }); err != nil {
+		logrus.WithError(err).WithField("request", req).Fatalln("Error while unmarshaling")
+	}
+
+	Expect(*respData.User.Name).To(Equal(testUpdatedUserName))
+	Expect(*respData.User.UniqueUserID).To(Equal(testUniqueUserId))
+	Expect(entity.UserType(respData.User.Type)).To(Equal(testUserType))
+}
+
+func getUser(port string) model.User {
+	// create a queryCollection (safe to share across requests)
+	client := graphql.NewClient(fmt.Sprintf("http://localhost:%s/query", port))
+
+	//@todo may need to make this more fault proof?! What if the test is executed from the root dir? does it still work?
+	b, err := os.ReadFile("../api/graphql/graph/queryCollection/user/listUsers.graphql")
+	Expect(err).To(BeNil())
+	str := string(b)
+	req := graphql.NewRequest(str)
+
+	req.Var("filter", map[string]string{"uniqueUserId": testUniqueUserId})
+	req.Var("first", 1)
+	req.Var("after", "0")
+
+	req.Header.Set("Cache-Control", "no-cache")
+	ctx := context.Background()
+
+	var respData struct {
+		Users model.UserConnection `json:"Users"`
+	}
+	if err := util2.RequestWithBackoff(func() error { return client.Run(ctx, req, &respData) }); err != nil {
+		logrus.WithError(err).WithField("request", req).Fatalln("Error while unmarshaling")
+	}
+	Expect(respData.Users.TotalCount).To(Equal(1))
+	return *respData.Users.Edges[0].Node
+}
+
+var _ = Describe("Creating and updating entity via API", Label("e2e", "Entities"), func() {
+	var s *server.Server
+	var cfg util.Config
+
+	BeforeEach(func() {
+		_ = dbm.NewTestSchema()
+
+		cfg = dbm.DbConfig()
+		cfg.Port = util2.GetRandomFreePort()
+		s = server.NewServer(cfg)
+
+		s.NonBlockingStart()
+	})
+
+	AfterEach(func() {
+		s.BlockingStop()
+	})
+
+	When("New user is created via API", func() {
+		var user model.User
+		BeforeEach(func() {
+			createUser(cfg.Port)
+			user = getUser(cfg.Port)
+		})
+		It("shall assign CreatedBy and CreatedAt metadata fields and shall keep nil in UpdatedBy, UpdatedAt and DeltedAt metadata fields", func() {
+			Expect(entity.UserType(user.Type)).To(Equal(testUserType))
+			Expect(user.Metadata).To(Not(BeNil()))
+			Expect(*user.Metadata.CreatedBy).To(Equal(testCreatedBy))
+
+			createdAt, err := time.Parse(dbDateLayout, *user.Metadata.CreatedAt)
+			Expect(err).Should(BeNil())
+			Expect(createdAt).Should(BeTemporally("~", time.Now().UTC(), 3*time.Second))
+
+			Expect(*user.Metadata.UpdatedBy).To(BeEmpty())
```

Comment:
Do we really check for `BeEmpty()`? in the `UpdatedBy` field?

Reply:
I wanted to highlight that UpdatedBy is not assigned when the object is created. In some solutions of such problems we see that modified/updated time is assigned by created time at creation of an object (eg. Linux File)